### PR TITLE
avoid recursive exec() of QgsCredentials

### PR DESCRIPTION
**Pull request by jef-n**
followup c9e76164982 - mutex didn't only guard the cache, but also the dialog

### CODE REVIEW

**Comment by nyalldawson** [comment]
@jef-n I don't understand the need for this -- and I believe it's going to bring back the original deadlock which my fix was designed to avoid.

Why would the dialog need a mutex? it's always run in the main thread

**Comment by jef-n** [comment]
If there are multiple layers eg. from the same postgres database requiring the same credentials, there will be multiple credential requests in parallel - but the second and following are canceled on exec() because the first request is still executing.

**Comment by nyalldawson** [comment]
Can you please test to make sure #20826 hasn't been revived?

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -1589,6 +1589,10 @@ QgisApp::~QgisApp()
   qDeleteAll( mCustomDropHandlers );
   qDeleteAll( mCustomLayoutDropHandlers );
 
+  // replace gui based network access managers with failing only ones
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: This part should not be revived

**Comment by elpaso** [comment]
Please have a look to my comments on https://issues.qgis.org/issues/21461

**Comment by giedrioks26** [comment]
this is related https://issues.qgis.org/issues/21582?

**Comment by jef-n** [comment]
> this is related https://issues.qgis.org/issues/21582?

yes.  The reverted change cause this

**Pull request merged**